File: SPOJ/ACPC11B.cpp
#include <iostream>
using namespace std;

int abs(int numb){
	if (numb >= 0){
		return numb;
	}
	else{
		return numb * -1;
	}
}

int main(){
	int T,N,M;
	int i,j,temp;
	cin >> T;
	while (T > 0){
		cin >> N;
		int N_arr[N];
		for (i = 0; i< N; i++){
			cin >> N_arr[i];
		}
		cin >> M;
		int M_arr[M];
		for(j = 0; j < M;j++){
			cin >> M_arr[j];
		}
		int min = 1000001;
		for ( i = 0; i < N; i ++){
			for(j = 0;j < M; j ++){
				temp = abs(M_arr[j] - N_arr[i]);
				if ( temp < min){
					min = temp;
				}
			}
		}
		cout  << min << endl;
		T--;
	}
	return 0;
}

//DONE

File: SPOJ/ACPC11B.py
import math
for _ in range(input()):
	first_array = map(int,raw_input().split())
	second_array = map(int,raw_input().split())
	min = 1000001
	for i in range(1,first_array[0]):
		for j in range(1,second_array[0]):
			temp = abs(second_array[j] - first_array[i])
			if temp < min:
				min = temp
	print min